### PR TITLE
Royalty swap

### PR DESCRIPTION
**Pull request by cartercarlson**

### CODE REVIEW

File: contracts/RoyaltyReceiver.sol [hunk 1]
```diff
@@ -3,27 +3,65 @@ pragma solidity 0.8.12;
...
     address public dao;
+    IERC20 public fluidToken;
+    IERC20 public weth;
+    address public pair;
+    IUniswapV2Router02 public router = IUniswapV2Router02(
+        0x1b02dA8Cb0d097eB8D57A175b88c7D8b47997506
+    );
+
+    constructor(
+        IFluidToken _fluidToken,
```
Comment (sacgrover) [inline contracts/RoyaltyReceiver.sol:25]: IFluidToken has not been imported to this contract

File: contracts/RoyaltyReceiver.sol [hunk 2]
```diff
@@ -3,27 +3,65 @@ pragma solidity 0.8.12;
...
+        // pre-approve router spending weth
+        weth.approve(address(router), type(uint256).max);
     }
 
-    function claimRoyalties(address token) external {
+    /// @notice Claim royalties earned from FLUID NFT market sales
+    /// @dev Swaps half the royalties to FLUID and sends to stakers
+    function claimRoyalties() external {
         // divide rewards by two - distribute 
         uint256 balance = IERC20(token).balanceOf(address(this));
```
Comment (sacgrover) [inline contracts/RoyaltyReceiver.sol:45]: `token ` is not defined in this contract

File: contracts/RoyaltyReceiver.sol [hunk 3]
```diff
@@ -3,27 +3,73 @@ pragma solidity 0.8.12;
...
+
+        swapWethForTokens(half);
+        IERC20(weth).transfer(dao, half);
+        IERC20(weth).transfer(msg.sender, functionCallReward);
+    }
+
+    function swapWethForTokens(uint256 amount) private {
+        address[] memory path = new address[](2);
+        path[0] = fluidToken;
+        path[1] = weth;
```
Comment (sacgrover) [inline contracts/RoyaltyReceiver.sol:55]: path addresses seem in wrong orders as we are swapping weth for fluid.. so first address should be for weth as input and 2nd address should be fluid token for output.

Comment (cartercarlson) [reply contracts/RoyaltyReceiver.sol:55]: :+1:

**Review by sacgrover — APPROVED** [review]

**Pull request merged**